x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/action/PreviewTransformAction.java

```diff
@@ -115,15 +121,22 @@ public TransformConfig getConfig() {
             return config;
         }
 
+        public boolean previewAsIndexRequest() {
+            return previewAsIndexRequest;
+        }
+
         @Override
         public void writeTo(StreamOutput out) throws IOException {
             super.writeTo(out);
             this.config.writeTo(out);
+            if (out.getTransportVersion().supports(PREVIEW_AS_INDEX_REQUEST)) {
+                out.writeBoolean(previewAsIndexRequest);
+            }
```

Comment:
Let's throw if somebody specifically requested `true`, but the transport version doesn't support it. This way they don't think they ran into a bug, but instead their cluster just isn't fully upgraded yet.